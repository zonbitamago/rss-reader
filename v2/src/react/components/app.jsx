'use strict';

import React from 'react';

class App extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <main>
        Hello World3!!
      </main>
    )
  }
}


module.exports = App
