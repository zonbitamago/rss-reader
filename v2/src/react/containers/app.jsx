'use strict';

import React from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import Side from '../components/side.jsx';
import MainPanels from '../components/mainPanels.jsx';
import styles from '../styles/app.css';
import * as actions from '../actions/index.js';

export class App extends React.Component {
  render() {
    const {mapstate, actions, store} = this.props;
    return (
      <main className={styles.main}>
        <Side actions={actions} rssListModal={mapstate.rssListModal.rssListModalOpen} urlList={mapstate.rssListModal.urlList} store={store}/>
        <MainPanels actions={actions}/>
      </main>
    )
  }
}

const mapState = (state, ownProps) => ({mapstate: state});

function mapDispatch(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  }
}

export default connect(mapState, mapDispatch)(App);
