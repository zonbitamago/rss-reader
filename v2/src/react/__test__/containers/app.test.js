'user strict';
jest.dontMock('../../containers/app.jsx');
import React from 'react';
import renderer from 'react-test-renderer';
import {App} from '../../containers/app.jsx';
import mockdate from 'mockdate';

const actions = {};

test('sample', function() {
  expect(1).toBe(1);
});

test('snapshot', function() {
  mockdate.set(1434319925275);
  const tree = renderer.create(<App actions={actions}/>).toJSON();
  expect(tree).toMatchSnapshot();

  mockdate.reset();
});
