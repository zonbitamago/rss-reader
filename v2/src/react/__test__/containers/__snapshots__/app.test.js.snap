// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<main
  className="main"
>
  <aside>
    <div
      className="ui small inverted vertical labeled icon ui left very thin visible sidebar menu"
    >
      <i
        aria-hidden="true"
        className="icon link item icon"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="red shutdown icon"
        />
      </i>
      <i
        aria-hidden="true"
        className="icon link item icon"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="refresh icon"
        />
      </i>
      <i
        aria-hidden="true"
        className="icon link item icon"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="yellow rss icon"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        />
      </i>
      <i
        aria-hidden="true"
        className="icon link item icon"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="green minus square icon"
        />
      </i>
      <i
        aria-hidden="true"
        className="icon link item icon"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="teal settings icon"
        />
      </i>
      <i
        aria-hidden="true"
        className="icon link item icon"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="olive github icon"
        />
      </i>
    </div>
    <div
      className="sidetime"
    >
      <time
        className="MMDD"
      >
        6/15
      </time>
      <br />
      <time
        className="HHmmss"
      >
        07:12:05
      </time>
    </div>
  </aside>
  <section
    className="mainPanels"
  >
    <div
      className="ui blue inverted pointing secondary head menu"
    >
      <div
        className="link item"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="home large icon"
        />
      </div>
      <div
        className="item"
        onClick={[Function]}
      >
        <h3
          className="ui inverted header"
        >
          Home
        </h3>
      </div>
    </div>
    <div
      className="itemPanel"
    />
  </section>
</main>
`;
